scripts/module/keybinder.nut

```diff
@@ -1,36 +1,41 @@
-keys <- {
-  "a" : 0x41,
-    "b" : 0x42,
-    "c" : 0x43,
-    "d" : 0x44,
-    "e" : 0x45,
-    "f" : 0x46,
-    "g" : 0x47,
-    "h" : 0x48,
-    "i" : 0x49,
-    "j" : 0x4A,
-    "k" : 0x4B,
-    "l" : 0x4C
-    }; // FIXME: Complete the table
-
-
+dofile('./keytable.nut');
 binds <- array(100, null);
 
+function extractFirstWord( input ) {
+  local ret = {};
+  local firstSpace = input.find(" ");
+  if (!firstSpace) {
+    firstSpace = input.len()-1;
+  } else {
+    ret.rawset("second", input.slice(firstSpace+1));
+  }
+  ret.rawset("first", input.slice(0, firstSpace));
+  return ret;
+}
+
 function onPlayerCommand( player, cmd, text ) {
   cmd = cmd.tolower();
   if (cmd == "b") {
     text = text.tolower();
     local bindTable = {};
-    local textArr = split(text, " ");
-    if (textArr.len() < 2) {
+    if (text.len() < 3) {
       MessagePlayer("Usage: /" + cmd + " <key> <command>", player);
-    } else if (!(textArr[0] in keys)) {
+      return;
+    }
+    local argTable = extractFirstWord(text);
+    local key = argTable.rawget("first");
+    if (!(key in keys) && !isNum(key)) {
```

Comment:
I'd do errors last, as fallback:

```
if key in key:
  ..
else if IsNum(key):
  // might be conflicting for inputs like "1" which is both, a number *and* a key. So you might want a prefix? say "*1" or ":1" or "key-1" or ...
  ..
else:
  error
```

Reply:
"`0`", "`1`", ..., "`9`" are treated like strings, "`09`" or "`0x0#`" is treated as key codes.
I'll swap the order. :+1:

Reply:
That doesn't sound like a good idea to me. I believe some scancodes will be up to 16 bit. I'd rather have a prefix of some sorts. "0x" would be fine, "#", etc.

Reply:
Ah, well, as long as squirrel thinks it's a number we'll be fine. Sorry for being vague.